modules/sdk-core/src/bitgo/wallet/wallet.ts

```diff
@@ -1624,12 +1657,15 @@ export class Wallet implements IWallet {
 
     const signTransactionParams = {
       ...presign,
-      txPrebuild,
+      txPrebuild: { ...txPrebuild, walletId: this.id() },
       pubs,
       coin: this.baseCoin,
     };
 
     if (_.isFunction(params.customSigningFunction)) {
+      if (this.baseCoin.usesUnspentModel()) {
+        return await this.signTransactionUtxoExternalSigner(params.customSigningFunction, signTransactionParams);
```

Comment:
we should try to move this to the coin-specific part

```
if (typeof this.baseCoin.signWithCustomSigningFunction === 'function') {
  return await this.baseCoin.signWithCustomSigningFunction(params.customSigningFunction, signTransactionParams);
}
```

Reply:
Done